api/src/main/java/org/apache/iceberg/Metrics.java

```diff
@@ -120,4 +124,90 @@ public Long recordCount() {
   public Map<Integer, ByteBuffer> upperBounds() {
     return upperBounds;
   }
+
+  /**
+   * Implemented the method to enable serialization of ByteBuffers.
+   * @param out The stream where to write
+   * @throws IOException On serialization error
+   */
+  private void writeObject(ObjectOutputStream out) throws IOException {
+    out.writeObject(rowCount);
+    out.writeObject(columnSizes);
+    out.writeObject(valueCounts);
+    out.writeObject(nullValueCounts);
+
+    writeByteBufferMap(out, lowerBounds);
+    writeByteBufferMap(out, upperBounds);
+  }
+
+  private static void writeByteBufferMap(ObjectOutputStream out, Map<Integer, ByteBuffer> byteBufferMap)
+      throws IOException {
+    if (byteBufferMap == null) {
+      out.writeInt(-1);
+
+    } else {
+      // Write the size
+      out.writeInt(byteBufferMap.size());
+
+      for (Map.Entry<Integer, ByteBuffer> entry : byteBufferMap.entrySet()) {
+        // Write the key
+        out.writeObject(entry.getKey());
+
+        // Write the value
+        if (entry.getValue() != null) {
+          // Copy the actual values from the buffer
+          ByteBuffer bb = entry.getValue();
+          byte[] bytes = new byte[bb.remaining()];
+          bb.get(bytes);
+          bb.position(bb.position() - bytes.length); // Restores the buffer position
```

Comment:
We have a `ByteBuffers` util located in the core module at `org.apache.iceberg.util.ByteBuffers` that handles the appropriate copying and restoration of ByteBuffers.

Please consider using that for consistency and for easier refactoring of ByteBuffer issues in the future. Here's a link to that class: https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/util/ByteBuffers.java

Reply:
I agree with @kbendick. You can call `ByteBuffers.toByteArray` instead of adding the logic to read here.

Reply:
Thanks @kbendick! Have not noticed that class.
Had to refactor that and move it to the API package to reuse it. Please check if you agree with the move, or you think there is a better place for it.

Thanks,
Peter